### PR TITLE
Fixing various workflow dropdown snags

### PR DESCRIPTION
**Pull request by DavidHollins6**
### 💬 Description
I'll comment on the code changes because there's multiple fixes in here!

### ✅ Checklist
- [ ] Tests written
- [ ] Browser tested
- [ ] Added to documentation
- [ ] Added to storybook

### CODE REVIEW

File: lib/Dropdown/DropdownContent.js [hunk 1]
```diff
@@ -100,20 +100,18 @@ export function DropdownContent({
...
+  const classNames = cx(`dropdown__content shadow ${className}`, {
+    'is-active': showContent,
+    'dropdown__content--collapse': collapse,
+    'dropdown__content--right': right,
+    'dropdown__content--centre': centre,
+    'dropdown__content--top': top,
+    'dropdown__content--fix-right': fixRight,
+    'dropdown__content--full': full,
+    'dropdown__content--noborder': noBorder,
+    'border border-solid border-neutral-90 rounded': !noBorder,
```
Comment (DavidHollins6) [inline lib/Dropdown/DropdownContent.js:112]: This must have been a recent fix. We had added some border tailwind classes but they always apply even when `noBorder` is set.

File: lib/src/components/workflowStep/workflowStepCollapsibleBody.js [hunk 2]
```diff
@@ -1,11 +1,25 @@
...
+      setShouldAllowOverflow(false);
+    }
+  }, [showBody]);
+
+  const onTransitionEnd = () => {
+    console.log(showBody, shouldAllowOverflow);
+    if (showBody) {
+      setShouldAllowOverflow(true);
+    }
+  };
```
Comment (DavidHollins6) [inline lib/src/components/workflowStep/workflowStepCollapsibleBody.js:22]: Ugh. There were some overflow issues around animations and dropdowns in the body. This code just hides the overflow while the animation is in progress, then shows any overflow afterwards ready for when dropdowns appear.

File: lib/src/prefabs/workflowDropdown/WorkflowDropdownContent.js [hunk 3]
```diff
@@ -3,7 +3,11 @@ import Dropdown from '../../../Dropdown';
 
 export function WorkflowDropdownContent({ children }) {
   return (
-    <Dropdown.Content noTransform className="workflow-dropdown__content">
+    <Dropdown.Content
+      noTransform
+      noBorder
```
Comment (DavidHollins6) [inline lib/src/prefabs/workflowDropdown/WorkflowDropdownContent.js:8]: We'd like to do our own border.

File: lib/src/prefabs/workflowDropdown/workflowDropdown.scss [hunk 4]
```diff
@@ -1,4 +1,11 @@
...
-  @apply p-0 text-left;
+  @apply p-0 text-left w-full border border-solid border-neutral-90 rounded;
+
+  .workflow-step {
+    @apply border-b border-t-0 border-l-0 border-r-0 border-solid border-neutral-90;
+  }
+
+  .workflow-step:last-of-type {
+    @apply border-0;
+  }
```
Comment (DavidHollins6) [inline lib/src/prefabs/workflowDropdown/workflowDropdown.scss:10]: We only want a bottom border on each step, except the last one.

**Review by kyleharper — APPROVED** [review]
Noice 👍

**Pull request merged**